conjure_oxide/tests/generated_tests.rs

```diff
@@ -299,44 +299,47 @@ fn integration_test_inner(
 }
 
 fn assert_vector_operators_have_partially_evaluated(model: &conjure_core::Model) {
-    model.constraints.transform(Arc::new(|x| {
-        use conjure_core::ast::Expression::*;
-        match &x {
-            Bubble(_, _, _) => (),
-            Atomic(_, _) => (),
-            Sum(_, vec) => assert_constants_leq_one(&x, vec),
-            Min(_, vec) => assert_constants_leq_one(&x, vec),
-            Max(_, vec) => assert_constants_leq_one(&x, vec),
-            Not(_, _) => (),
-            Or(_, vec) => assert_constants_leq_one(&x, vec),
-            And(_, vec) => assert_constants_leq_one(&x, vec),
-            Eq(_, _, _) => (),
-            Neq(_, _, _) => (),
-            Geq(_, _, _) => (),
-            Leq(_, _, _) => (),
-            Gt(_, _, _) => (),
-            Lt(_, _, _) => (),
-            SafeDiv(_, _, _) => (),
-            UnsafeDiv(_, _, _) => (),
-            SumEq(_, vec, _) => assert_constants_leq_one(&x, vec),
-            SumGeq(_, vec, _) => assert_constants_leq_one(&x, vec),
-            SumLeq(_, vec, _) => assert_constants_leq_one(&x, vec),
-            DivEqUndefZero(_, _, _, _) => (),
-            Ineq(_, _, _, _) => (),
-            // this is a vector operation, but we don't want to fold values into each-other in this
-            // one
-            AllDiff(_, _) => (),
-            WatchedLiteral(_, _, _) => (),
-            Reify(_, _, _) => (),
-            AuxDeclaration(_, _, _) => (),
-            UnsafeMod(_, _, _) => (),
-            SafeMod(_, _, _) => (),
-            ModuloEqUndefZero(_, _, _, _) => (),
-            Neg(_, _) => (),
-            Minus(_, _, _) => (),
-        };
-        x.clone()
-    }));
+    let expressions = &model.constraints;
+    for expression in expressions {
+        expression.transform(Arc::new(|x| {
```

Comment:
this doesn't return anything, so I think you can also use universe instead of transform. see this for an example: https://github.com/conjure-cp/conjure-oxide/blob/3dc68ec5fd90ec32666352c909efc3204ce22d95/crates/conjure_core/src/rules/normalisers/bool.rs#L85

Reply:
or even `model.constraints.universe_bi`. feel free to ignore this, we can do it in a separate PR. just fix the test files though :)